packages/core/src/application/application-factory.ts

```diff
@@ -99,8 +99,8 @@ class AppFactory {
             container,
             appTypeOrMiddlewares as Array<express.RequestHandler>,
           );
-          return app as AppExpress;
```

Comment:
I can't approve this request to remove the `return app as IApplication` within each switch block, The reason why is because we will have multiple interfaces, as commented in the code you can see that we can possibly have a fastify interface, and we will have much more in the future.

The return has to be constrained within each switch case.

Reply:
Yes, it makes sense, but I think there is a way to have an only one return statement rather than multiple ones, but anyways I will keep it like before.

Reply:
Hi @mohanadft of course there is, there are 2 strategies I can think of it:

1 - Using a helper function to reduce the repetition
```typescript
private static async createApp<T extends AppExpress | AppFastify>(
  container: Container,
  AppType: new () => T,
  appTypeOrMiddlewares: (new () => T) | Array<express.RequestHandler | Handler>,
): Promise<T> {
  let app: T = {} as T;

  if (this.isOpinionated(appTypeOrMiddlewares)) {
    app = container.resolve(AppType);
    await app.create(container);
  } else {
    app = container.get<T>(AppType);
    await app.create(container, appTypeOrMiddlewares as Array<express.RequestHandler | Handler>);
  }
  return app;
}

public static async create(
  container: Container,
  appTypeOrMiddlewares:
    | (new () => AppExpress)
    | (new () => AppFastify)
    | Array<express.RequestHandler | Handler>,
  httpServerFactory?: new () => AppExpress | AppFastify,
): Promise<IApplicationExpress | IApplicationFastify | AppExpress | AppFastify> {
  if (httpServerFactory) {
    return this.createApp(container, httpServerFactory, appTypeOrMiddlewares);
  }
  return this.createApp(container, AppExpress, appTypeOrMiddlewares);  // Default to AppExpress
}
```

2 - my favorite one, generics and type inference

```typescript
public static async create<T extends AppExpress | AppFastify>(
  container: Container,
  appTypeOrMiddlewares:
    | (new () => T)
    | Array<express.RequestHandler | Handler>,
  httpServerFactory?: new () => T,
): Promise<T> {
  let app: T = {} as T;
  
  const AppType = httpServerFactory || AppExpress;  // Default to AppExpress
  
  if (this.isOpinionated(appTypeOrMiddlewares)) {
    app = container.resolve(AppType);
    await app.create(container);
  } else {
    app = container.get<T>(AppType);
    await app.create(container, appTypeOrMiddlewares as Array<express.RequestHandler | Handler>);
  }

  return app;
}
```

The question is, why am I not focusing in simplifying these things? Answer is simple, too many things in my plate, as well as I am focusing on having a stable platform first then when I have enough time I can simplify the code as well as improve performance as much as we can.

As I gave you these two tips, try to implement the second one, test, verify it works then I can accept your complete change in this PR. How that sounds? :)

Reply:
Great, since you merged it, should I make a new PR?